packages/auth/README.md

```diff
@@ -307,33 +308,288 @@ console.log(context.currentUser)
 
 You can map the "raw decoded JWT" into a real user object by passing a `getCurrentUser` function to `createCreateGraphQLHandler`
 
-Our recommendation is to create a `src/lib/auth.js|ts` file that exports a `getCurrentUser` function (You may already have a stub function.)
+Our recommendation is to create a `src/lib/auth.js|ts` file that exports a `getCurrentUser` as well as a `hasRole` function (You may already have stub functions.)
 
 ```js
-import { getCurrentUser } from 'src/lib/auth'
+import { getCurrentUser, hasRole } from 'src/lib/auth'
 // Example:
 //  export const getCurrentUser = async (decoded) => {
 //    return await db.user.findOne({ where: { decoded.email } })
 //  }
-``
+//
+// Example for hasRole and RBAC
+//
+//  const appMetadata = (decoded) => {
+//    return decoded[`https://example.com/app_metadata`] || {}
+//  }
+//
+//  const roles = (decoded) => {
+//    return appMetadata(decoded).authorization?.roles || []
+//  }
+//
+//  export const getCurrentUser = async (decoded) => {
```

Comment:
I know we can't add types here, but is there a nice way to add some comments to show the shape of this function? e.g. mine looks like this in TS

```
interface TokenHeader {
  type: 'cli' | 'jwt'
  token: string
}

export const getCurrentUser = async (
  decodedToken: string,
  tokenHeader: TokenHeader
) => {
  let user
```

Reply:
☝️ Definitely a question best answered by @peterp

Reply:
I've been dealing with this exact problem when creating serverless functions, which currently look like this:
```js
export const handler = (event, context, callback) => {
  return { statusCode: '404' }
}
```
Type-safety be damned! Unless I define those types myself... I don't know what's going on!?

One pattern that I've stumbled upon is to introduce a wrapping function:
```js
export const handler = createServerlessFunction(({event, context })=> {
 return {
    statusCode: 200,
    body: JSON.stringify('pew pew pew'),
 }
})
```

This is nice, because I can provide types for event and context, and can specify exactly what the user can return! I can also introduce some lifecycle things into this function that give me as a framework creator a bit more control. It's not nice because it's a bit redundant for just adding types.

Maybe, maybe, maybe we can introduce a wrapper for `getCurrentUser`? (These are always optional if you "understand" the structure that needs to be exported.

Reply:
No related @peterp but if you wanted typescript magic in JS, have you tried adding `// @ts-check` to the top of the file? VSCode picks this up and does type checking if it can infer things.

One idea for this function is to use JS doc style comments

```
** /**
 * Handle 
 * @param decoded a decoded token on the type of Auth Provider used. Custom auth just returns the token
 * @param tokenDetails ...
* @example ......
**//**
```

Reply:
@dac09 or @peterp Anything actionable for me here w/ re: to Token header and TS?

Reply:
How do you guys feel about adding a commented out the js doc style?

Reply:
@dac09 @dthyresson Nothing actionable from my side, but I really like @dac09's jsdoc suggestion! 🧠

Reply:
JS doc style comments .. yup! That's a good habit I should get into again.

Reply:
I don't think we _have_ to add that in here... it could be done in another PR.

Reply:
Or we can just add an issue and we can discuss it further.

Reply:
Yeah makes sense!

Reply:
Ok, I already added it ... I am just reworking all the docs and then testing. Have to step away for a bit but will commit soon.

Reply:
1. Strange now seeing an error in CI that not seeing local. I'll have to investigate.

```
3:41:43 PM - Project 'tsconfig.json' is out of date because output file '.buildcache/AuthProvider.d.ts' does not exist

3:41:43 PM - Building project '/home/runner/work/redwood/redwood/packages/auth/tsconfig.json'...

##[error]src/AuthProvider.tsx(137,36): error TS2339: Property 'roles' does not exist on type 'CurrentUser'.
##[error]src/AuthProvider.tsx(198,11): error TS2322: Type '(role: string) => any' is not assignable to type '() => boolean'.
```

2. I am also going to setup an app w/ Netlify Identity and try out role access